src/Tasks/Microsoft.NET.Build.Tasks/DependencyContextBuilder2.cs

```diff
@@ -0,0 +1,859 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text;
+using Microsoft.Extensions.DependencyModel;
+using NuGet.Packaging;
+using NuGet.Packaging.Core;
+using NuGet.ProjectModel;
+using NuGet.Versioning;
+
+namespace Microsoft.NET.Build.Tasks
+{
+    internal class DependencyContextBuilder2
```

Comment:
nit: The "style" is close to existing DependencyContextBuilder. The methods are still long. It could use some extract method at least. Considering there are so many fields, it should be able to extract more classes. I don't think we should block on that, but I hope this class can be more clear. And ideally add some unit tests

Reply:
I also agree with Will that the complexity of some of these methods inherited from the old implementation might need some refactoring for clarity.

If now isn't the time to do that, then I think we should file an issue to address it when we remove the old code.